examples/c/5_series/ahrs/README.md

```diff
@@ -0,0 +1,187 @@
+- **Manual Parsing**: Direct parsing of incoming MIP packets using the MIP parser
+- **Callback Functions**: Function pointer-based callbacks for data processing
+- **Explicit Memory Management**: Manual buffer and resource management
+- **Type Safety**: Uses C structs and enums for data type safety
+- **Error Codes**: Command results handled through return codes and error checking
+
+## C Implementation Features
+
+This example showcases:
+- **MIP Interface**: Core C interface for device communication (`mip_interface`)
+- **Serial Port Management**: Low-level serial port operations
+- **Parser Integration**: Direct use of the MIP packet parser
+- **Memory Safety**: Careful buffer management and bound checking
+- **Portability**: Cross-platform compatibility (Windows/Unix)
+
+## Custom Communication Functions
+
+The example implements custom communication handlers:
+- **Send Function**: `mip_interface_user_send_to_device()` handles outgoing data
+- **Receive Function**: `mip_interface_user_recv_from_device()` manages incoming data
+- **Timeout Handling**: Configurable timeouts for reliable communication
+
+## Filter Data Output
+
+The example streams the following filter data:
+
+### TOW Data
+- **Units**: seconds
+- **Description**: Time of Week - GPS time reference
+- **Format**: Floating-point timestamp value
+
+### Euler Angles Data
+- **Units**: radians
+- **Description**: Orientation expressed as Euler angles
+- **Format**: [Roll, Pitch, Yaw] angle vector
+
+## Streaming Output Format
+
+The example displays filter data in the following format:
+```
+TOW = 123456.789     Euler Angles = [ 0.012345, -0.067890,  1.234567]
+```
+
+## Usage
+
+1. Connect your 5-series AHRS device to the specified serial port
+2. Update the `PORT_NAME` constant if using a different port
+3. Compile and run the example
+4. Follow the gyro bias capture prompt (keep the device stationary)
+5. The program will:
+    - Initialize the device
+    - Configure data streaming
+    - Wait for the filter to reach AHRS mode
+    - Stream data for the specified runtime
+    - Display filter state transitions
+    - Display real-time filter data
+    - Clean up and exit
+
+## Filter State Progression
+
+The example monitors and displays filter state transitions:
+1. **Startup** - Filter startup
+2. **Initialization** - Filter initialization
+3. **Run Solution Valid** - Valid filter solution
+4. **Run Solution Invalid** - Invalid filter solution
+
+## Error Handling
+
+The example includes comprehensive error handling with:
+- Command result checking using `mip_cmd_result`
+- Connection failure detection and recovery
+- Graceful termination functions for different error types
+- Detailed error messages with context
+
+## C Features
+
+This example demonstrates:
+- Standard C11 programming practices
+- Minimal external dependencies
+- Direct hardware interface programming
+- Efficient memory usage patterns
+- Cross-platform serial communication
+
+## Building
```

Comment:
For this and all other example READMEs, this should also be pretty high up. I would put this right after Usage and Configuration.